src/shapes/circle.class.ts

```diff
@@ -1,201 +1,213 @@
+  },
+
+  /**
+   * Sets radius of an object (and updates width accordingly)
+   * @return {Circle} thisArg
+   */
+  setRadius: function(value) {
+    this.radius = value;
+    return this.set('width', value * 2).set('height', value * 2);
+  },
+
+  /**
+   * Returns object representation of an instance
+   * @param {Array} [propertiesToInclude] Any properties that you might want to additionally include in the output
+   * @return {Object} object representation of an instance
+   */
+  toObject: function(propertiesToInclude) {
+    return this.callSuper('toObject', ['radius', 'startAngle', 'endAngle'].concat(propertiesToInclude));
+  },
+
+  /* _TO_SVG_START_ */
+
+  /**
+   * Returns svg representation of an instance
+   * @return {Array} an array of strings with the specific svg representation
+   * of the instance
+   */
+  _toSVG: function() {
+    const angle = (this.endAngle - this.startAngle) % 360;
+
+    if (angle === 0) {
+      return [
+        '<circle ', 'COMMON_PARTS',
+        'cx="0" cy="0" ',
+        'r="', this.radius,
+        '" />\n',
+      ];
+    } else {
+      const {radius} = this;
+      const start = degreesToRadians(this.startAngle),
+        end = degreesToRadians(this.endAngle),
+        startX = fabric.util.cos(start) * radius,
+        startY = fabric.util.sin(start) * radius,
+        endX = fabric.util.cos(end) * radius,
+        endY = fabric.util.sin(end) * radius,
+        largeFlag = angle > 180 ? '1' : '0';
+      return [
+        '<path d="M ' + startX + ' ' + startY,
+        ' A ' + radius + ' ' + radius,
+        ' 0 ', + largeFlag + ' 1', ' ' + endX + ' ' + endY,
+        '" ', 'COMMON_PARTS', ' />\n',
+      ];
+    }
+  },
+  /* _TO_SVG_END_ */
+});
+
+/* _FROM_SVG_START_ */
+/**
+ * List of attribute names to account for when parsing SVG element (used by {@link Circle.fromElement})
+ * @static
+ * @memberOf Circle
+ * @see: http://www.w3.org/TR/SVG/shapes.html#CircleElement
+ */
+Circle.ATTRIBUTE_NAMES = ['cx', 'cy', 'r', ...fabric.SHARED_ATTRIBUTES];
+
+/**
+ * Returns {@link Circle} instance from an SVG element
+ * @static
+ * @memberOf Circle
+ * @param {SVGElement} element Element to parse
+ * @param {Function} [callback] Options callback invoked after parsing is finished
+ * @param {Object} [options] Partial Circle object to default missing properties on the element.
+ * @throws {Error} If value of `r` attribute is missing or invalid
+ */
+Circle.fromElement = function(element, callback, options = {}) {
+  const parsedAttributes = fabric.parseAttributes(element, Circle.ATTRIBUTE_NAMES);
+
+  if (!isValidRadius(parsedAttributes)) {
+    throw new Error('value of `r` attribute is required and can not be negative');
+  }
 
-})(typeof exports !== 'undefined' ? exports : window);
+  const left = (parsedAttributes.left || 0) - parsedAttributes.radius;
+  const top = (parsedAttributes.top || 0) - parsedAttributes.radius;
+
+  callback(new fabric.Circle({
+    ...options, ...parsedAttributes,
```

Comment:
Went with the approach of keeping options consistent with Rect in this pr, it now accepts and correctly uses options.

Reply:
this may have unintended consequences. options are the values present in the svg tag, and are meant to either pass in width/height or other small things.
I wouldn't add them here.